app/apollo/resolvers/cluster.js

```diff
@@ -181,7 +180,9 @@ const clusterResolvers = {
           throw new NotFoundError(context.req.t('Could not find the cluster with name {{clusterName}}.', {'clusterName':clusterName}), context);
         }
 
-        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context, [cluster.id, cluster.name, clusterName]);
+        logger.info({req_id, user, org_id, clusterName}, `${queryName} validating`);
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context, [cluster.cluster_id, clusterName]);
+        logger.info({req_id, user, org_id, clusterName}, `${queryName} validating - authorized`);
```

Comment:
- The `validating` log message should be at the beginning of the validation section (i.e. right after the `try {` line).
- Simple checks for valid input should be first
- While the DB query can be before the `validAuth` call, the throwing of NotFoundError should be _after_, to avoid information leakage.
- Similarly it shouldn't return "more than one" error until after it has determined if the user is authorized or not -- if the user isn't authorized on what they are querying, we shouldn't tell them there exist multiple.
- The validAuth call will need to change it's `[ id, name ]` argument depending on whether the cluster was found in the DB or not.